### ISSUE TITLE
Multiple @font-face rules

### ISSUE THREAD

**Issue by suburbiaboy** [wkhtmltopdf/wkhtmltopdf#2435]
Hopefully this might shed some light issues others have had with multiple @font-face instances not working correctly. For example, if you have a setup like:

    @font-face {
      font-family: FooReg
      font-weight: normal;
      src: url('../fonts/FooReg.otf');
    }
    @font-face {
      font-family: FooBold
      font-weight: normal;
      src: url('../fonts/FooBold.otf');
    }

You may find only 'FooReg' works in the PDF. The reason/workaround is that both share the same Font Family name (Foo) despite having different names and filenames.

If you you open the font in FontForge you can change the Font family name to be different for each and they will work as expected in PDF.

**Comment by ashkulz** [comment]
@suburbiaboy: so this is more of a solution than a real issue, right?

**Comment by suburbiaboy** [comment]
@ashkulz – It is bug that the font's Family Name is being used, it could cause all sorts of unexpected results. The family name is irrelevant in CSS normally. Editing a font file to make it display correctly is not ideal but it is also not too difficult, so perhaps this is low priority.

**Comment by ashkulz** [comment]
Is this handled better in the `0.13` version?

**Comment by JustOnce** [comment]
It's looks like I have the same issue, but I've opened fonts in FontForge and they have all different family names (or may be I do something wrong). Example html/css files and rendered pdf here https://github.com/JustOnce/wkhtmltopdf_test. 

$ wkhtmltopdf -V
wkhtmltopdf 0.12.2-6a13a51 (with patched qt)  

$ lsb_release -a
LSB Version:	core-2.0-amd64:core-2.0-noarch:core-3.0-amd64:core-3.0-noarch:core-3.1-amd64:core-3.1-noarch:core-3.2-amd64:core-3.2-noarch:core-4.0-amd64:core-4.0-noarch:core-4.1-amd64:core-4.1-noarch:security-4.0-amd64:security-4.0-noarch:security-4.1-amd64:security-4.1-noarch
Distributor ID:	LinuxMint
Description:	Linux Mint 17 Qiana
Release:	17
Codename:	qiana

Also the same issue on server with Ubuntu 12.04.

Any offer/help please? Look's like wkhtmltopdf uses for all paragraphs a font that was first mentioned in a document for paragraph. 

p.s. In this case which type of font will be used for render? woff, ttf?
@font-face {
  font-family: 'RussianRailGPro-Bold';
  src: url('../fonts/RussianRailGPro-Bold.eot');
  src: url('../fonts/RussianRailGPro-Bold.eot?#iefix') format('embedded-opentype'),
       url('../fonts/RussianRailGPro-Bold.woff2') format('woff2'),
       url('../fonts/RussianRailGPro-Bold.woff') format('woff'),
       url('../fonts/RussianRailGPro-Bold.ttf') format('truetype'),
       url('../fonts/RussianRailGPro-Bold.svg#RussianRailGPro-Bold') format('svg');
}

**Comment by suburbiaboy** [comment]
Don't forget to add `font-weight: normal;` to each `@font-family`. I think I've seen mention of issues if you don't.

**Comment by suburbiaboy** [comment]
Also I saw an issue with `font-family` names that were longer than 8 characters. I think that was fixed but maybe try to make the first 8 chars unique?

**Comment by richardolsson** [comment]
I just want to confirm that we are getting this issue as well. The only solution that seems to work is, like @suburbiaboy suggests, to change the family name in the font file (e.g. using FontForge). We've also tried the following, to no avail:

* Using only woff, woff2, or ttf font files.
* Using base64 encoded inlined font files in the CSS.
* Change names of files
* Use short font-family names (CSS) without the family prefix (e.g. "Bold" instead of "Roboto-Bold")

**Comment by alehro** [comment]
This approach also helped me. But the only change of the family name didn't help. So, I changed everything in FontForge's Element->FontInfo page, including versions.

**Comment by michaelvillar** [comment]
This definitely still happens and fixed it using this trick too

**Comment by pier** [comment]
Just confirming that is still very much an issue. As confirmed by @richardolsson and @pixelmachine, the only way around this is by modifying the font itself (notably, the font family name) using something like TransType. This is of course far from ideal. Is there anyone on the dev team taking this on board?

**Comment by jonathanasquier** [comment]
Hi, I'm using @pixelmachine's technique, it works when only one font-family is called in the style sheet, but when setting a second font-family in another class they both fail. Any idea?
Thanks!

**Comment by pixelmachine** [comment]
I've had success using numerous different fonts @jonathanasquier in the same PDF.

What always works for me:
1. Pick an 8 or less char name for each font and variation e.g. `HelvBold`, `HelvLite`, `TimeReg`
2. Every variation needs to be a separate font file, with all the details renamed in Font Forge using the short name determined before.
3. Give the font files the same short name (e.g. `HelvBold.ttf`)
4. Setup the font-family with the same:

    @font-face {
      font-family: HelvBold
      font-weight: normal;
      src: url('../fonts/HelvBold.ttf');
    }
    @font-face {
      font-family: HelvLite
      font-weight: normal;
      src: url('../fonts/HelvLite.ttf');
    }
    @font-face {
      font-family: TimeReg
      font-weight: normal;
      src: url('../fonts/TimeReg.ttf');
    }

To be honest no.2 is the key but keeping it consistent makes it easier to deal with.

**Comment by dennisschaaf** [comment]
I made a little fontforge script to rename fonts: https://gist.github.com/dennisschaaf/a0256c8460aedb2a4ee1e01443722ec2

**Comment by sbatschelet** [comment]
wkhtmltopdf 0.12.3 (with patched qt) I can only get a single font to work even using the rename method.

**Comment by sbatschelet** [comment]
@pixelmachine can you please share a version that is working for you with the above setup.

**Comment by pixelmachine** [comment]
@sbatschelet I'm running 0.12.2.1 (with patched qt)

**Comment by besciualex** [comment]
Had the same issue but solved it in a simpler way: 

I had multiple fonts in the same family (EnzoOffc), with different names. I have different font for each combination of **font-weight** and **font-style** properties. Take a look bellow:

```
/*Thin - ITALIC*/
@font-face {
    font-family: 'EnzoOffc';
    src: url('/public/fonts/EnzoOffc-LightItalic.eot');
    src: local('☺'),
    url('/public/fonts/EnzoOffc-LightItalic.woff') format('woff'),
    url('/public/fonts/EnzoOffc-LightItalic.ttf') format('truetype'),
    url('/public/fonts/EnzoOffc-LightItalic.svg') format('svg');
    font-weight: 200;
    font-style: italic;
}
/*Normal*/
@font-face {
    font-family: 'EnzoOffc';
    src: url('/public/fonts/EnzoOffc-Medium.eot');
    src: local('☺'),
    url('/public/fonts/EnzoOffc-Medium.woff') format('woff'),
    url('/public/fonts/EnzoOffc-Medium.ttf') format('truetype'),
    url('/public/fonts/EnzoOffc-Medium.svg') format('svg');
    font-weight: normal;
    font-style: normal;
}
/*Normal - ITALIC*/
@font-face {
    font-family: 'EnzoOffc';
    src: url('/public/fonts/EnzoOffc-MediumItalic.eot');
    src: local('☺'),
    url('/public/fonts/EnzoOffc-MediumItalic.woff') format('woff'),
    url('/public/fonts/EnzoOffc-MediumItalic.ttf') format('truetype'),
    url('/public/fonts/EnzoOffc-MediumItalic.svg') format('svg');
    font-weight: normal;
    font-style: italic;
}
/*Bold*/
@font-face {
    font-family: 'EnzoOffc';
    src: url('/public/fonts/EnzoOffc-Bold.eot');
    src: local('☺'),
    url('/public/fonts/EnzoOffc-Bold.woff') format('woff'),
    url('/public/fonts/EnzoOffc-Bold.ttf') format('truetype'),
    url('/public/fonts/EnzoOffc-Bold.svg') format('svg');
    font-weight: bold;
    font-style: normal;
}
body{
     font-family: 'EnzoOffc';
}
```

As you can see I used the same **font-family** for all, and played only with **font-style** and **font-weight** properties, like I exemplified bellow:

```
<h1 style="font-weight:bold;">Bold Title which uses EnzoOffc-Bold.ttf</h1>
<p style="font-weight:normal;font-style:italic;">Normal Italic Text which uses EnzoOffc-MediumItalic.ttf</p>
```
Using this solution you avoid using **Font-Forge**, and **wkhtmltopdf** sees all font variations. Hope it saves some time for you folks.

**Comment by g4b0** [comment]
I'm having the same issue. With the following code just the first font-face is rendered, while in browser also bold and italic are ok:

```css
@font-face {
    font-family: 'MyriadPro';
    src:  url('pdf-fonts/MyriadPro-Regular.otf') format('truetype');
}
@font-face {
    font-family: 'MyriadPro';
    src:  url('pdf-fonts/MyriadPro-Bold.otf') format('truetype');
    font-weight: bold;
}
@font-face {
    font-family: 'MyriadPro';
    src:  url('pdf-fonts/MyriadPro-It.otf') format('truetype');
    font-style: italic;
}
@font-face {
    font-family: 'MyriadPro';
    src:  url('pdf-fonts/MyriadPro-BoldIt.otf') format('truetype');
    font-weight: bold;
    font-style: italic;
}
```

**Comment by pixelmachine** [comment]
@g4b0 Did you edit them in Font Forge as detailed above? Also try:

1. Change the file name to a short name (no idea if this makes any difference but I always do it just in case.)
2. Make font-style and font-weight normal. If you're using a bold/italic font there's no need to specify that.

**Comment by besciualex** [comment]
@g4b0 First of all, use Font Forge to edit all the fonts you need. Don't try anything else, it would be a waste of time. I had 8 fonts using the same font-family, but different font-weight and font-style attributes. 
What you need to do:
- rename all your physical font names to have 8 chars maximum;
- use (as now) truetype format
In Font Forge:
- make sure that you have the same font-family to all fonts
- in FontForge > Element > Font Info, in the next tabs: PS Names, OS/2 (all sub-tabs), TTF Names update the properties to reflect 100% the real aspect of fonts. 

Example:
1. Rename MyriadPro-BoldIt.otf  ==> myprboit.otf
2. In Font Forge, PS Names tab update Family Name: to "MyriadPro" (common for all fonts)
3. in OS/2 tab update Weight Class to 700 (the one for bold), Style Map to "italic"
4.in TTF Names add property "Preffered Family" = "MyriadPro" (common for all fonts)
... (do this for all fonts above)
9) in CSS play with variations as you did, to make the difference between fonts.
```
@font-face {
    font-family: 'MyriadPro';
    src:  url('pdf-fonts/myprboit.otf') format('truetype');
    font-weight: bold;
    font-style: italic;
}

```

**Comment by PhilterPaper** [comment]
If these (unmodified) fonts display correctly in browsers, then this behavior should be reported to WebKit as a bug in their code (assuming it's not something in the surrounding wkHTMLto* code). Modifying font files may be a workaround, but it's not a desirable solution.

**Comment by pixelmachine** [comment]
Pretty sure it's not WebKit since they do work in all browsers. Could be QtWebKit I suppose. Might be worth a look through their issues before assuming it's wkhtmltox.

**Comment by g4b0** [comment]
These unmodified fonts display correctly in browsers, so I suppose it's a wkHTMLtopdf bug. Maybe the workaround can solve my problem, but I prefer to wait if a cleaner solution comes out.

@ashkulz any hint?

**Comment by besciualex** [comment]
@g4b0 **wkHTMLtopdf** works very good. It is font's manufacturer fault that they set wrong properties. To test my affirmation, try the same code you wrote using a Windows font (from Fonts directory). You will also notice that all fonts have maximum 8 chars in name. 

P.S. Don't forget to import it like you did on MyriadPro, even if your browser knows how to render it, wkHTMLtopdf doesn't know without import.

All the best.

**Comment by pixelmachine** [comment]
That's crazy. These fonts (I have tried many) work fine on all compatible systems and browsers. The only place they don't work is with WKHTMLTOx. 

In fact you can check the [official MS OpenType spec](https://www.microsoft.com/typography/otspec/otvaroverview.htm) and you'll see there's no such 8 char limitation or recommendation.

The bug is in WKHTMLTOx or QtWebKit or the combination of both.

**Comment by g4b0** [comment]
@besciualex what do you mean by "Don't forget to import it like you did on MyriadPro"? I never imported it on my local computer, I've just putted it into pdf-fonts/ directory and used it with 
```css
src:  url('pdf-fonts/MyriadPro-Regular.otf') format('truetype');
```

**Comment by pixelmachine** [comment]
@g4b0 just noticed you're using an OTF but you have format set as 'truetype'. Try removing format (I never specify that) or setting it to 'opentype'.

    src:  url('pdf-fonts/myprboit.otf');
    src:  url('pdf-fonts/myprboit.otf') format('opentype');

Probably won't make a difference but worth limiting the number or possible issues.

**Comment by pixelmachine** [comment]
Ok @ashkulz I've found the root of the problem – WebKit, or at least an old version of it used in QtWebKit. [This WebKit bug report](https://bugs.webkit.org/show_bug.cgi?id=29433) describes the exact issue (from 2009!).

Suggested workaround is exactly as originally described above:
> As a work-around one can edit each font in a font family and change its family name to be different.

The conclusion (from 2012) is:
> It is fixed in trunk WebKit. Qt 4.8.2 still comes shipped with QtWebKit 2.2, which was branched off from trunk before that.

So a custom build would be necessary.

**Comment by besciualex** [comment]
@g4b0 I meant strictly when you try a font from Fonts folder. Let's say you use Segoe UI font on a page. This font exists by default in all Windows machines, so any browser installed on Windows won't have any problem rendering it, because they have it installed. But an external software on a UNIX machine don't have it and reads it only if is imported using `@font-face{}`.

Regarding the 8 char maximum name length restriction, go to _Control Panel\All Control Panel Items\Fonts\Segoe UI_, copy all fonts from there and paste them on desktop. Read there their real and limited names. I'm not saying that this is 100% mandatory. 
I'm saying that I tested many fonts, and only the ones who had all the font properties correctly set (all from Microsoft, directory Fonts) and limited file names, work good in WKHTMLTOPDF. I used this as a 'guide' to edit my Enzo fonts properties and file names. Only after this step it worked as expected.

**TL;DR** always use `@font-face{ }` when you use fonts which needs to be rendered by WKHTMLTOPDF, even Arial Black.

**Comment by pixelmachine** [comment]
@besciualex I reckon the guy at Microsoft who puts the fonts in Windows [is still using DOS](https://en.wikipedia.org/wiki/8.3_filename) ;)

Anyway I agree with your suggestions, that'll get it working.

**Comment by ArthurianX** [comment]
I think this should be put up top somewhere in a "Known Issues" section as I spent 14 hours finding out ContentNotFoundError1 exit was because of this.

**Comment by EcoFreak** [comment]
As mentioned in #3248 I'm having a problem with my fonts in Ubuntu.

I currently have this in my sass file, what do I need to do to my font files and sass file in order to have everything working:
```


@font-face {
    font-family: 'titillium_bdbold';
    src: url('/fonts/titillium-bold-webfont.eot');
    src: url('/fonts/titillium-bold-webfont.eot?#iefix') format('embedded-opentype'),
         url('/fonts/titillium-bold-webfont.woff2') format('woff2'),
         url('/fonts/titillium-bold-webfont.woff') format('woff'),
         url('/fonts/titillium-bold-webfont.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
}

@font-face {
    font-family: 'titilliumlight';
    src: url('/fonts/titillium-light-webfont.eot');
    src: url('/fonts/titillium-light-webfont.eot?#iefix') format('embedded-opentype'),
         url('/fonts/titillium-light-webfont.woff2') format('woff2'),
         url('/fonts/titillium-light-webfont.woff') format('woff'),
         url('/fonts/titillium-light-webfont.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
}

@font-face {
    font-family: 'titilliumregular';
    src: url('/fonts/titillium-regular-webfont.eot');
    src: url('/fonts/titillium-regular-webfont.eot?#iefix') format('embedded-opentype'),
         url('/fonts/titillium-regular-webfont.woff2') format('woff2'),
         url('/fonts/titillium-regular-webfont.woff') format('woff'),
         url('/fonts/titillium-regular-webfont.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
}

@font-face {
    font-family: 'titilliumbold';
    src: url('/fonts/titillium-semibold-webfont.eot');
    src: url('/fonts/titillium-semibold-webfont.eot?#iefix') format('embedded-opentype'),
         url('/fonts/titillium-semibold-webfont.woff2') format('woff2'),
         url('/fonts/titillium-semibold-webfont.woff') format('woff'),
         url('/fonts/titillium-semibold-webfont.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
}


@font-face {
  font-family: 'Gill Sans MT Pro Light';
  src: url('/fonts/GillSansMTPro-Light.eot');
  src: url('/fonts/GillSansMTPro-Light.eot?#iefix') format('embedded-opentype'),
  url('/fonts/GillSansMTPro-Light.woff') format('woff'),
  url('/fonts/GillSansMTPro-Light.ttf') format('truetype');
  font-weight: 300;
  font-style: normal;
}

@font-face {
  font-family: 'Gill Sans MT Pro Medium';
  src: url('/fonts/GillSansMTPro-Medium.eot');
  src: url('/fonts/GillSansMTPro-Medium.eot?#iefix') format('embedded-opentype'),
  url('/fonts/GillSansMTPro-Medium.woff') format('woff'),
  url('/fonts/GillSansMTPro-Medium.ttf') format('truetype');
  font-weight: 500;
  font-style: normal;
}
```

Thanks :)

**Comment by pixelmachine** [comment]
@EcoFreak, both I and @besciualex have written full step by step instructions above, have you tried either?

**Comment by ArthurianX** [comment]
As an addendum to the steps above check to see if you have any missing files, e.g. the woff2 file. Personally, after I did everything from the above I had a missing SVG file, and I would still call it in the CSS and the PDF conversion would still fail.

**Comment by PhilterPaper** [comment]
In #3248, @EcoFreak reported that simply adding the font weights to the CSS fixed the problem (in his case).

**Comment by besciualex** [comment]
@PhilterPaper There might be two possibilities: he already had good fonts (with good file names and font properties), or he simply may think it's working because wkhtmltopdf actually bolds, lights the fonts, and it doesn't really use the correct font.
Let me explain: I had 8 fonts, and they were all variations of semilight, light, normal, bold, and boldest. I tought I did it first, but then the designer came and asked me to use only one bold font on the PDF => the result? it rendered perfect. Then he asked me to print a PDF where we had all variations of font. The bold font used among all others was not rendered as the one (alone) earlier. That's how he proved me he was right.

**Comment by EcoFreak** [comment]
@besciualex I was using the adobe acrobat pdf reader and checking out the fonts tab in the properties for the generated file. I'm sure it's using the font (because it's visible noticible and is also specified in the fonts tab) and I didin't change any other configuration regarding font name or properties.

**Comment by besciualex** [comment]
@EcoFreak I'm glad it worked for you on the first try. 👍

**Comment by maikusch** [comment]
can someone say whats wrong on my font-face?
my pdf does not display any font.
`@font-face {
 font-family: "Arial";
 src: url(data:font/truetype;charset=utf-8;base64, ...) format("truetype");
font-weight: normal;
font-style: normal;`

if i remove font-face everything runs fine. But then the font is shitty.

**Comment by PhilterPaper** [comment]
According to https://www.paulirish.com/2009/bulletproof-font-face-implementation-syntax/ (a few years old!), `format("truetype");` is optional and can be dropped (for TT). You might try that.

I'm not sure if you want a `;` or a `,` after the **format**, and I'll assume you have a closing `}` on this.

**Comment by clintconklin** [comment]
Ran into this today with various weights/styles of OpenSans. Tried the renaming of the font family in fontforge, but no love there.

Ended up using these font definitions...

https://github.com/FontFaceKit/open-sans/blob/gh-pages/open-sans.css

...and then css declarations like:

```css
.regular {
	font-family: 'Open Sans';
}

.bold-italic {
	font-family: 'Open Sans';
	font-weight: bold;
	font-style: italic;
}

.semibold {
	font-family: 'Open Sans';
	font-weight: 600;
}
```

etc. Worked like a charm.

**Comment by rruchte** [comment]
I ran into this issue when trying to use multiple Gotham faces in the same file. The font files all had unique family names, which threw me off for a while. It turns out there was a field in the TTF Names table called "Preferred Family" that was common to all of the files. Deleting that field from each file solved the problem. I wrote up some more details in a [stack overflow answer](https://stackoverflow.com/questions/36427422/wkhtmltopdf-multiple-font-faces-styles/45524377#45524377).

**Comment by landsman** [comment]
What about improve this library rather some special editing font-face using etc?

**Comment by hkraji** [comment]
First of all let me say, it is nobody's job to maintain this library and thanks to all the devs for making it possible to use this. Do not like it, fix it yourself.

That being said I had the same problem, on version `0.12.3` and `0.12.5` (latest). I did the following and it helped so bear with me.

1. Rename files (to use 8 char maximum names e.g. Regular)
2. Open fonts with an editor (I used FontLab7) and changed Name, FamilyName, FullName to "Regular"
3. Set here weight class to `normal` aka `regular`
4. Font face looks like this (used .ttf)

```
@font-face {
  font-family: Regular;
  font-weight: normal;
  // rails code to insert base64 encoding
  src: url(<%= asset_data_base64('ibm_plex/Regular.ttf') %>);
}
```

5. CSS that uses it looks like this

```
.fw-400, .fw-500 {
  font-family: Regular;
  font-weight: normal;
  font-style: normal;
}
```

I know similar things were said above, but here is the final cut so you do not have to think too much. God knows I did to make this work 😄 Good luck! ❤️

**Comment by akaspick** [comment]
Renaming the font-family for each style in fontforge finally fixed this issue for me.  I specify the font-weight's for each font and then specify the appropriate font-family for each place I need the different style fonts.

```
@font-face {
    font-family: "Segoe UI Regular";
    font-weight: 500;
    src: url(<%= asset_data_base64('Segoe UI/SegoeUI-Regular-Custom.ttf') %>);
}
@font-face {
    font-family: "Segoe UI Semibold";
    font-weight: 600;
    src: url(<%= asset_data_base64('Segoe UI/SegoeUI-Semibold-Custom.ttf') %>);
}
@font-face {
    font-family: "Segoe UI Bold";
    font-weight: 700;
    src: url(<%= asset_data_base64('Segoe UI/SegoeUI-Bold-Custom.ttf') %>);
}
```

In my css:

```
body {
  font-size: 10pt;
  font-family: 'Segoe UI Regular';
}

h1, h2, h3 {
  text-transform: uppercase;
  font-family: 'Segoe UI Bold';
}
```

So far, so good.

**Comment by michaelotoole** [comment]
Just ran into this problem, tried everything above but it wasn't working as was only changing the PS Names in font forge. 

Just incase anyone else runs into the same problems, you may need to change the font names in the TTF Names section in FontForge too (the preferred English US was still using the old font family name for my problem font).

**Comment by umesh2k8** [comment]
> I've had success using numerous different fonts @jonathanasquier in the same PDF.
> 
> What always works for me:
> 
> 1. Pick an 8 or less char name for each font and variation e.g. `HelvBold`, `HelvLite`, `TimeReg`
> 2. Every variation needs to be a separate font file, with all the details renamed in Font Forge using the short name determined before.
> 3. Give the font files the same short name (e.g. `HelvBold.ttf`)
> 4. Setup the font-family with the same:
> 
> ```
> @font-face {
>   font-family: HelvBold
>   font-weight: normal;
>   src: url('../fonts/HelvBold.ttf');
> }
> @font-face {
>   font-family: HelvLite
>   font-weight: normal;
>   src: url('../fonts/HelvLite.ttf');
> }
> @font-face {
>   font-family: TimeReg
>   font-weight: normal;
>   src: url('../fonts/TimeReg.ttf');
> }
> ```
> 
> To be honest no.2 is the key but keeping it consistent makes it easier to deal with.


```
@font-face {
    font-family: "regularfnt";
    src: url("/u01/Umesh_Rnd/helvetica/final/helvreg-webfont.woff") format("woff");
    font-weight: normal;
    font-style: normal;
}

@font-face {
    font-family: "boldfnt";
    src: url("/u01/Umesh_Rnd/helvetica/final/helvbld-webfont.woff") format("woff");
    font-weight: normal;
    font-style: normal;
}
@font-face {
    font-family: "bolditalicfnt";
     src: url("/u01/Umesh_Rnd/helvetica/final/helvbld-webfont.woff") format("woff");
	font-weight: normal;
    font-style: italic;
}
@font-face {
    font-family: "italicfnt";
    src: url("/u01/Umesh_Rnd/helvetica/final/helvita-webfont.woff") format("woff");
    font-weight: normal;
    font-style: normal;
}
```

It worked for me

**Comment by john-n-smith** [comment]
FontForge wouldn't work for me. Instead I installed the command line tool `ttx` (via `brew install fonttools` on OS X).

Then:
- `ttx /path/to/font.otf`, this outputs `/path/to/font.ttx`
- Open `/path/to/font.ttx` using a text editor
- Remove the following name elements:
  - `<namerecord nameID="1" platformID="1"...>`
  - `<namerecord nameID="1" platformID="3"...>`
  - `<namerecord nameID="16" platformID="3"...>`
- Save the file
- `ttx /path/to/font.ttx`, this outputs `/path/to/font#1.otf`
- Use `/path/to/font#1.otf` in place of `/path/to/font.otf`

I found no need to set `font-weight` etc. properties on my CSS, choose a name <= 8 chars long, rename the font filename.

Interestingly I did not experience this issue using v0.12.5 on Alpine, but I did on 0.12.6 on Amazon Linux and Debian Buster.

**Comment by Ledmonds** [comment]
> Hopefully this might shed some light issues others have had with multiple @font-face instances not working correctly. For example, if you have a setup like:
> 
> ```
> @font-face {
>   font-family: FooReg
>   font-weight: normal;
>   src: url('../fonts/FooReg.otf');
> }
> @font-face {
>   font-family: FooBold
>   font-weight: normal;
>   src: url('../fonts/FooBold.otf');
> }
> ```
> 
> You may find only 'FooReg' works in the PDF. The reason seems to be that both share the same Font Family name ("Foo") despite having different Font names ("FooReg", "FooBold") and filenames.
> 
> If you you open the font in FontForge you can edit the Font's family name to be different for each variation and they will work as expected in PDF.

Life Saver xD

**Comment by logankeenan** [comment]
This thread was super helpful! I was having font issues when consuming wkhtmltopdf in `wkhtmltopdf-binary (0.12.6.5)` and  `wkhtmltopdf-heroku (2.12.6.0)`


I had to give each font a unique name and change the styles to normal with FontForge even if the font style was `bold` or `font-weight: 500`.  Then declare the `font-style: normal;` and `font-weight: normal;` within the CSS.  The example below demonstrates how to update and use a bold font.

Open FontForge and changing the following:
* TTF Name -> (US) Preferred Styles -> Change to regular from Bold
* TTF Name -> (US) Preferred Family -> Change to _Front Name Bold_ from _Font Name_
* OS/2  -> Weight Class -> Change to 400 Regular from Bold
* PS Names -> Weight -> Change to Regular from Bold


CSS:
```css
@font-face {
  font-family: "Font Name";
  font-weight: normal;
  font-style: normal;
  src: url("font-name-modified.otf");
}

@font-face {
  font-family: "Font Name Bold";
  font-style: normal;
  font-weight: normal;
  src: url("font-name-bold-modified.otf");
}

body {
  font-family: "Font Name";
}

.bold {
  font-weight: normal;
  font-family: "Font Name Bold";
}

h1 { 
  font-weight: normal;
  font-family: "Font Name Bold";
}

```

*Note:* I didn't have to limit the font name by 8 characters or any similar limitation.

**Comment by BrnPer** [comment]
> This thread was super helpful! I was having font issues when consuming wkhtmltopdf in `wkhtmltopdf-binary (0.12.6.5)` and `wkhtmltopdf-heroku (2.12.6.0)`
> 
> I had to give each font a unique name and change the styles to normal with FontForge even if the font style was `bold` or `font-weight: 500`. Then declare the `font-style: normal;` and `font-weight: normal;` within the CSS. The example below demonstrates how to update and use a bold font.
> 
> Open FontForge and changing the following:
> 
> * TTF Name -> (US) Preferred Styles -> Change to regular from Bold
> * TTF Name -> (US) Preferred Family -> Change to _Front Name Bold_ from _Font Name_
> * OS/2  -> Weight Class -> Change to 400 Regular from Bold
> * PS Names -> Weight -> Change to Regular from Bold
> 
> CSS:
> 
> ```css
> @font-face {
>   font-family: "Font Name";
>   font-weight: normal;
>   font-style: normal;
>   src: url("font-name-modified.otf");
> }
> 
> @font-face {
>   font-family: "Font Name Bold";
>   font-style: normal;
>   font-weight: normal;
>   src: url("font-name-bold-modified.otf");
> }
> 
> body {
>   font-family: "Font Name";
> }
> 
> .bold {
>   font-weight: normal;
>   font-family: "Font Name Bold";
> }
> 
> h1 { 
>   font-weight: normal;
>   font-family: "Font Name Bold";
> }
> ```
> 
> **Note:** I didn't have to limit the font name by 8 characters or any similar limitation.

Worked perfectly! Thank you a lot!
